dataplane/vppagent/pkg/converter/memif_interface_converter.go

```diff
@@ -36,23 +37,39 @@ func NewMemifInterfaceConverter(c *connection.Connection, name string) Converter
 	return rv
 }
 
+func NewMemifInterfaceWithIpConverter(c *connection.Connection, name string, ipAddress string) Converter {
+	rv := &MemifInterfaceConverter{
+		Connection: c,
+		name:       name,
+		ipAddress:  ipAddress,
+	}
+	return rv
+}
+
 func (c *MemifInterfaceConverter) ToDataRequest(rv *rpc.DataRequest) (*rpc.DataRequest, error) {
 	if rv == nil {
 		rv = &rpc.DataRequest{}
 	}
 	socketFilename, ok := c.Mechanism.Parameters[connection.SocketFilename]
 	if !ok {
-		socketFilename = "mymemif.sock"
+		socketFilename = "memif.sock"
 	}
-	socketPath := path.Join(c.Mechanism.Parameters[connection.Workspace], socketFilename)
+	socketPath := path.Join(memif.BuildMemifDirectory(c.Mechanism), socketFilename)
 	isMaster, err := strconv.ParseBool(c.Mechanism.Parameters[connection.Master])
 	if err != nil {
 		isMaster = false
 	}
+
+	var ipAddresses []string
+	if c.ipAddress != "" {
```

Comment:
Just to confirm my understanding, in the vppagent-dataplane case, you simply don't pass in an ipAddress, and thus don't assign one to the vpp side (correct behavior) and in the nse case, you do pass in an ipAddress and thus *do* assign it on the vpp side (also correct behavior)? :)